Sources/Core/UseCase/SnackbarGetButtonTypeUseCase.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  SnackbarGetButtonTypeUseCase.swift
+//  SparkSnackbar
+//
+//  Created by louis.borlee on 03/09/2024.
+//  Copyright © 2024 Adevinta. All rights reserved.
+//
+
+import SparkButton
+
+// sourcery: AutoMockable
+protocol SnackbarGetButtonTypeUseCasable {
+    func execute(
+        intent: SnackbarIntent,
+        variant: SnackbarVariant
+    ) -> SnackbarButtonType
+}
+
+final class SnackbarGetButtonTypeUseCase: SnackbarGetButtonTypeUseCasable {
+    func execute(
+        intent: SnackbarIntent,
+        variant: SnackbarVariant
+    ) -> SnackbarButtonType {
+        switch intent {
+        case .success:
+            return self.getSuccessButtonType(variant: variant)
+        case .alert:
+            return self.getAlertButtonType(variant: variant)
+        case .error:
+            return self.getErrorButtonType(variant: variant)
+        case .info:
+            return self.getInfoButtonType(variant: variant)
+        case .neutral:
+            return self.getNeutralButtonType(variant: variant)
+        case .main:
+            return self.getMainButtonType(variant: variant)
+        case .basic:
+            return self.getBasicButtonType(variant: variant)
+        case .support:
+            return self.getSupportButtonType(variant: variant)
+        case .accent:
+            return self.getAccentButtonType(variant: variant)
+        case .surfaceInverse:
+            return self.getSurfaceInverseButtonType(variant: variant)
+        }
+    }
+
+    private func getSuccessButtonType(variant: SnackbarVariant) -> SnackbarButtonType {
+        let buttonIntent = ButtonIntent.success
+        let buttonVariant: ButtonVariant
```

Comment:
If you want, to limit the number of lines, you can : 
```swift
let buttonVariant: ButtonVariant = switch variant {
        case .filled: .filled
        case .tinted: .tinted
        }
```

OR

```swift
return .init(
    intent: ButtonIntent.alert,
    variant: switch variant {
        case .filled: .filled
        case .tinted: .tinted
        }
)
```

Reply:
Done